pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +78,32 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList); err != nil {
+		return err
+	}
+
+	for i := range ncList.Items {
+		stored := ncList.Items[i].DeepCopy()
+		if ncList.Items[i].Labels[v1beta1.NodePoolLabelKey] != np.Name {
+			continue
+		}
+
+		if ncList.Items[i].Annotations[v1beta1.NodePoolHashVersionAnnotationKey] != v1beta1.NodePoolHashVersion && ncList.Items[i].StatusConditions().GetCondition(v1beta1.Drifted) == nil {
+			ncList.Items[i].Annotations = lo.Assign(ncList.Items[i].Annotations, map[string]string{
+				v1beta1.NodePoolHashAnnotationKey:        np.Hash(),
+				v1beta1.NodePoolHashVersionAnnotationKey: v1beta1.NodePoolHashVersion,
+			})
+			if !equality.Semantic.DeepEqual(stored, ncList.Items[i]) {
+				if err := c.kubeClient.Patch(ctx, &ncList.Items[i], client.MergeFrom(stored)); err != nil {
+					return client.IgnoreNotFound(err)
```

Comment:
This means that we will exit out of this loop early returning a nil error in the case that one of the NodeClaims that we went to grab is NotFound, which isn't what we want here

Reply:
I'm wondering if we should just best-effort try to update everything and then create multi-err and return that. The other option is to just shift the `err != nil` check in the `if` statement to a `client.IgnoreNotFound(err) != nil` check

Reply:
I added a best-effort multi-err approach